exdb/views.py

```diff
@@ -46,34 +42,55 @@ def get_form_class(self):
             return ExperienceSaveForm
 
 
-class HallStaffDashboardView(TemplateView):
-    access_level = 'basic'
-    template_name = 'exdb/hallstaff_dash.html'
+class HallStaffDashboardView(ListView):
+    acess_leve = 'basic'
+    template_name = 'exdb/home.html'
+    context_object_name = 'experiences'
+
+    def get_queryset(self):
+        experience_approvals = ExperienceApproval.objects.filter(
+            approver=self.request.user, experience__status='ad')
+        experiences = Experience.objects.filter(Q(next_approver=self.request.user) | Q(
+            pk__in=experience_approvals.values('experience')))
+        return experiences
 
     def get_context_data(self):
         context = super(HallStaffDashboardView, self).get_context_data()
-        context['user'] = get_user_model().objects.prefetch_related(
-            'approval_queue__author',
-            'approval_queue__keywords',
-            'approval_queue__recognition',
-            'approval_set__experience__author',
-            'approval_set__experience__keywords',
-            'approval_set__experience__recognition'
-        ).get(pk=self.request.user.pk)
+        context['user'] = self.request.user
+
+        status_to_display = [_('Pending Approval'), _('Needs Evaluation'), _('Approved')]
+        experience_dict = OrderedDict()
+        for status in status_to_display:
+            experience_dict[status] = []
+        for experience in context[self.context_object_name]:
+            if experience.needs_evaluation():
+                experience_dict[_('Needs Evaluation')].append(experience)
+            else:
+                if experience.get_status_display() in experience_dict:
+                    experience_dict[experience.get_status_display()].append(experience)
+        context['experience_dict'] = experience_dict
+
+        one_week = timezone.now() + timezone.timedelta(days=7)
+        upcoming = []
+        for experience in context['experience_dict'][_('Approved')]:
+            if experience.start_datetime > timezone.now() and experience.start_datetime < one_week:
+                upcoming.append(experience)
+        context['upcoming'] = upcoming
         return context
 
 
 class RAHomeView(ListView):
+    template_name = 'exdb/home.html'
     access_level = 'basic'
-    template_name = 'exdb/ra_home.html'
     context_object_name = 'experiences'
 
     def get_queryset(self):
-        return Experience.objects.filter(author=self.request.user).order_by('created_datetime')
+        return Experience.objects.filter(Q(author=self.request.user) | (Q(planners=self.request.user) & ~Q(
+            status='dr'))).exclude(status__in=('ca')).order_by('created_datetime')
```

Comment:
If the author of an experience also selects themselves as a planner this page will display the experience twice.  Adding .distinct() here fixes that.